NullPointerException in HytalePermissionsProvider.write during Service Account Provisioning

When using the Automatic Provisioning feature (placing a .serviceaccount.json file in the provisioning folder), the Hytale server crashes with a NullPointerException whenever a player attempts to join or when the server triggers a syncSave() on permissions.json.

The crash occurs because a null UUID key is being injected into the global PermissionsModule user map.

**Stack Trace:**
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.UUID.toString()" because the return value of "java.util.Map$Entry.getKey()" is null
        at com.hypixel.hytale.server.core.permissions.provider.HytalePermissionsProvider.write(HytalePermissionsProvider.java:302)
        at com.hypixel.hytale.server.core.util.io.BlockingDiskFile.syncSave(BlockingDiskFile.java:75)


**Findings:**
The issue appears to be in WebServerPlugin.importServiceAccount. Based on the behavior, if the serviceAccountCredentialStore does not yet contain a mapping for the provided account name, the following logic likely proceeds with a null UUID:

    The plugin reads the JSON service account.

    It attempts to retrieve a UUID for the account name.

    If the account is new, it passes a null value to PermissionsModule.get().addUserToGroup(uuid, group).

    The Hytale Core API accepts the null key into its internal Map<UUID, UserData>.

    When HytalePermissionsProvider.write iterates over the map entries to save to disk, entry.getKey().toString() throws the NPE.

**Steps to Reproduce:**
    - Create a fresh Hytale server (Version 2026.01.15-c04fdfe10).
    - Install the nitrado:webserver plugin.
    - Create mods/Nitrado_WebServer/provisioning/test.serviceaccount.json with standard content (Name, PasswordHash, Groups, Permissions).
    - Start the server.
    - Attempt to join the server as a player.
    - The player is disconnected with the message: Exception when adding player to world!.

**Environment:**
    - Hytale Version: 2026.01.15-c04fdfe10
    - Plugin Version: 1.0.0 (Nitrado WebServer)
    - Java Version: 25

--- a/src/main/java/net/nitrado/hytale/plugins/webserver/WebServerPlugin.java
+++ b/src/main/java/net/nitrado/hytale/plugins/webserver/WebServerPlugin.java
@@ -9,7 +9,6 @@ import com.hypixel.hytale.server.core.plugin.JavaPluginInit;
 import com.hypixel.hytale.server.core.plugin.PluginBase;
 import com.hypixel.hytale.server.core.util.Config;
 import jakarta.servlet.Filter;
-import jakarta.servlet.http.HttpFilter;
 import jakarta.servlet.http.HttpServlet;
 import net.nitrado.hytale.plugins.webserver.authentication.AuthProvider;
 import net.nitrado.hytale.plugins.webserver.authentication.internal.AuthFilter;
@@ -338,27 +337,11 @@ public final class WebServerPlugin extends JavaPlugin {
         return this.webServer.getRegisteredPlugins();
     }
 
-    UUID createServiceAccount(String name, String password) throws IOException {
-        UUID uuid = UUID.randomUUID();
-
-        if (!name.startsWith("serviceaccount.")) {
-            name = "serviceaccount." + name;
-        }
-
-        try {
-            this.serviceAccountCredentialStore.setUserCredential(uuid, name, password);
-            PermissionsModule.get().addUserToGroup(uuid, "SERVICE_ACCOUNT");
-            return uuid;
-
-        } catch (IOException e) {
-            getLogger().at(Level.SEVERE).log("failed to create service account credentials: %s", e.getMessage());
-            throw e;
-        }
+    UUID createServiceAccountBcrypt(String name, String passwordHash) throws IOException {
+        return this.createServiceAccountBcrypt(UUID.randomUUID(), name, passwordHash);
     }
 
-    UUID createServiceAccountBcrypt(String name, String passwordHash) throws IOException {
-        UUID uuid = UUID.randomUUID();
-
+    UUID createServiceAccountBcrypt(@Nonnull UUID uuid, String name, String passwordHash) throws IOException {
         if (!name.startsWith("serviceaccount.")) {
             name = "serviceaccount." + name;
         }
@@ -381,7 +364,7 @@ public final class WebServerPlugin extends JavaPlugin {
         var name = document.getString("Name");
 
         // Delete the service account every time to also reset its permissions and groups
-        this.deleteServiceAccount(name);
+        var uuid = this.deleteServiceAccount(name);
 
         var enabled = document.getBoolean("Enabled");
         if (!enabled) {
@@ -389,8 +372,12 @@ public final class WebServerPlugin extends JavaPlugin {
         }
 
         var passwordHash = document.getString("PasswordHash");
-        this.createServiceAccountBcrypt(name, passwordHash);
-        var uuid = this.serviceAccountCredentialStore.getUUIDByName(name);
+
+        if (uuid == null) {
+            uuid = this.createServiceAccountBcrypt(name, passwordHash);
+        } else {
+            this.createServiceAccountBcrypt(uuid, name, passwordHash);
+        }
 
         var groups = document.getList("Groups", String.class);
         var permissions = document.getList("Permissions", String.class);
@@ -426,13 +413,18 @@ public final class WebServerPlugin extends JavaPlugin {
         }
     }
 
-    void deleteServiceAccount(String name) throws IOException {
+    UUID deleteServiceAccount(String name) throws IOException {
+        if (!name.startsWith("serviceaccount.")) {
+            name = "serviceaccount." + name;
+        }
+
         var uuid = this.serviceAccountCredentialStore.getUUIDByName(name);
         if (uuid == null) {
-            return;
+            return null;
         }
 
         this.deleteServiceAccount(uuid);
+        return uuid;
     }
 
     private WebServer getWebServer() {

